datafusion/sql/src/interval.rs

```diff
@@ -64,17 +64,28 @@ pub(crate) fn parse_interval(leading_field: &str, value: &str) -> Result<ScalarV
             }
 
             match interval_type.to_lowercase().as_str() {
-                "year" => Ok(align_interval_parts(interval_period * 12_f32, 0.0, 0.0)),
-                "month" => Ok(align_interval_parts(interval_period, 0.0, 0.0)),
+                "century" | "centuries" => {
+                    Ok(align_interval_parts(interval_period * 1200_f32, 0.0, 0.0))
```

Comment:
It might be nice in the future to name constants like `1200_f32` with some symbolic names

like 

```
let YEARS_PER_CENTRY: f32 = 1200_f32;
```

But that is just a stylistic thing -- it isn't like the number of years in a century is going to be changed at any point in our lifetimes 🤣